### PR TITLE
isEligible should fire on route change

### PR DESCRIPTION
**Pull request by japboy**
I need this Optimize module and want to run different experiments on each page, so I made #3 possible.

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -5,14 +5,23 @@ import experiments from '<%= options.experimentsDir %>'
 const MAX_AGE = <%= options.maxAge %>
 
 export default function (ctx, inject) {
+  ctx.app.router.beforeEach((to, from, next) => {
```
Comment (tschut) [inline lib/plugin.js]: Won't this override any existing function on beforeEach? Just wondering.

Comment (japboy) [reply lib/plugin.js]: `beforeEach` is a register that adds callback function, so it won't override existing functions, i think.

https://github.com/vuejs/vue-router/blob/v3.0.2/src/index.js#L121-L123

**Comment by japboy** [comment]
I think I need more considerations before this is accepted, so I close this PR. Sorry for bothering.

**Comment by tschut** [comment]
@japboy what's the problem? I'd really like to have this functionality too so would like to help if possible.

**Comment by japboy** [comment]
@tschut first of all, running multiple experiments at once is not ideal way of A/B testing or MVT. second, changing cookie format is necessary to support multiple experiment data to be stored properly. current implementation supports only one experiment at once. and i need some minor fixes if this is merged (like firing multiple route events at once etc...).

i think i should define MVT for multiple endpoint experiments instead of firing route event.

**Pull request closed without merging**